src/checks/level-exp.js

```diff
@@ -0,0 +1,21 @@
+module.exports = {
+  description: "A Pokémon's experience should match its level.",
+  field: 'level',
+  test (pkmn) {
+    const currentLevel = pkmn.level;
+    const currentExp = pkmn.exp;
+    const growthRate = pkmn.growthRate;
+    const maxExp = {
+      'slow-then-very-fast': 600000,
+      fast: 800000,
+      medium: 1000000,
+      'medium-slow': 1059860,
+      slow: 1250000,
+      'fast-then-very-slow': 1640000};
+
+    if (currentLevel === 100) {
+      return maxExp[growthRate] > currentExp;
+    }
+    return currentLevel < 100;
```

Comment:
Minor nitpick: Can you make the level a filter instead? Since this check only applies to Pokémon at level 100, it doesn't need to be run against Pokémon of any other level.

You can make it a filter by adding something like this:

```js
module.exports = {
  // ...
  filter: {level: 100},
  test (pkmn) {
    // ...
  }
};
```

Then you can remove the `pkmn.level` logic in the `test` function since it will only get run when the level is 100.